### PR TITLE
Add metrics observation

### PR DESCRIPTION
**Pull request by flomillot**

### CODE REVIEW

**Comment by flomillot** [comment]
I think we should add a default provider or make it mandatory in the endpoints specifications

File: src/main/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisWorkerService.java [hunk 1]
```diff
@@ -172,7 +181,8 @@ private CompletableFuture<SensitivityAnalysisResult> runSensitivityAnalysisAsync
                 return null;
             }
             SensitivityAnalysisParameters sensitivityAnalysisParameters = buildParameters(context);
-            Network network = getNetwork(context.getNetworkUuid(), context.getVariantId());
+            Network network = sensitivityAnalysisObserver.observe("network.load", context, () ->
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisWorkerService.java:184]: Remove this metric

**Review by klesaulnier — APPROVED** [review]

File: src/main/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisObserver.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    public SensitivityAnalysisObserver(@Value("${sensitivity-analysis.default-provider}") String defaultProvider,
+                                       @NonNull ObservationRegistry observationRegistry,
+                                       @NonNull MeterRegistry meterRegistry) {
+        this.defaultProvider = defaultProvider;
+        this.observationRegistry = observationRegistry;
+        this.meterRegistry = meterRegistry;
+    }
+
+    public <E extends Throwable> void observe(String name, SensitivityAnalysisRunContext runContext, Observation.CheckedRunnable<E> callable) throws E {
+        createObservation(name, runContext).observeChecked(callable);
```
Comment (sBouzols) [inline src/main/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisObserver.java:45]:
```suggestion
    public <E extends Throwable> void observe(String name, SensitivityAnalysisRunContext runContext, Observation.CheckedRunnable<E> runnable) throws E {
        createObservation(name, runContext).observeChecked(runnable);
```

Comment (flomillot) [reply src/main/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisObserver.java:45]: I just copy paste from other MS https://github.com/gridsuite/security-analysis-server/blob/6bab1554756090fef76bf5ec82dd3d6600fb4fd5/src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisObserver.java#L44-L46
But indeed it should be changed everywhere.

Comment (flomillot) [reply src/main/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisObserver.java]: https://github.com/gridsuite/security-analysis-server/pull/106

File: src/main/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisWorkerService.java (hunk 1 above)
Comment (flomillot) [reply src/main/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisWorkerService.java]: Ok, done

File: src/main/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisObserver.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            .lowCardinalityKeyValue(PROVIDER_TAG_NAME, provider)
+            .lowCardinalityKeyValue(TYPE_TAG_NAME, COMPUTATION_TYPE);
+    }
+
+    private void incrementCount(SensitivityAnalysisRunContext runContext, SensitivityAnalysisResult result) {
+        String provider = runContext.getProvider() != null ? runContext.getProvider() : defaultProvider;
+        Counter.builder(COMPUTATION_COUNTER_NAME)
+            .tag(PROVIDER_TAG_NAME, provider)
+            .tag(TYPE_TAG_NAME, COMPUTATION_TYPE)
+            .tag(STATUS_TAG_NAME, getStatusFromResult(result))
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisObserver.java:70]: Remove this

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisObserver.java (hunk 3 above)
Comment (flomillot) [reply src/main/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisObserver.java:70]: You mean I should not use incrementCount as in security server ?

Comment (flomillot) [reply src/main/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisObserver.java]: As discussed, I always return OK

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**